src/item_functions.cpp

```diff
@@ -35,4 +39,23 @@ bool can_be_unloaded( const item &itm )
     return itm.ammo_remaining() > 0 || itm.casings_count() > 0;
 }
 
+// TODO: Add consideration for BIONIC_GUNS when their fake_items get USES_BIONIC_POWER
+int shots_remaining( const Character &who, const item &it )
+{
+    units::energy energy_drain = units::from_kilojoule( it.get_gun_ups_drain() );
+    units::energy power = units::from_kilojoule( who.charges_of( itype_UPS ) );
+    if( it.ammo_required() > 0 && it.get_gun_ups_drain() > 0 ) {
+        // Both UPS and ammo, lower is limiting.
+        return std::min( it.ammo_remaining(), power / energy_drain );
+    } else if( it.get_gun_ups_drain() > 0 ) {
+        return power / energy_drain;
+    } else if( it.ammo_required() > 0 ) {
+        return it.ammo_remaining();
+    } else {
+        // Effectively infinite ammo.
+        return 10;
+    }
+    return 0;
```

Comment:
```suggestion
    }
    // Effectively infinite ammo.
    return 10;
```
`return 0;` is a dead code and will never be used.

Reply:
No, this is me not understanding if else correctly, one moment.